src/spatialnetworks.cpp

```diff
@@ -103,7 +103,7 @@ arma::mat join_spatiallines_coords(SEXP sldf, double startx, double starty) {
   double prevx = startx;
   double prevy = starty;
 
-  for (unsigned int i = 0; i < lines.length(); i++) {
+  for (int i = 0; i < lines.length(); i++) {
```

Comment:
It just makes the data types consistent. lines.length() is apparently a signed integer (which I wouldn't expect) so it should be compared to an integer (rather than an unsigned integer).